cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -397,7 +397,7 @@ class device_row_comparator {
         }
 
         if (lcol.num_child_columns() == 0) {
-          return cuda::std::pair(weak_ordering::EQUIVALENT, depth);
+          return cuda::std::pair(weak_ordering::EQUIVALENT, std::numeric_limits<int>::max());
```

Comment:
This is a bug. Previously this bug didn't have any chance to show up. Now with the new decomposition algorithm it will.

The logic is:
 * If we hit a null: return current `depth`.
 * If there is not any nulls, the returned `depth` must be ignored at the caller. Thus, `numeric_limit::max` is returned (so it will always be ignored).

Reply:
Ah I see this is the case where there are no nulls so far and no further children. And I guess before we weren't recursing far enough to ever hit that case. OK cool good find!